src/seasons.js

```diff
@@ -20,21 +26,37 @@ const type2Class = {
 
 const getType = (type, $) => {
   const result = []
-  const classToSearch = `.js-seasonal-anime-list-key-${type2Class[type]} .seasonal-anime.js-seasonal-anime`
+
+  // If TV has been selected, filter New from Continuing.
+  const typeString = matchSorter(possibleTypes, type)[0]
+
+  let classToSearch = `.js-seasonal-anime-list-key-${type2Class[typeString]} .seasonal-anime.js-seasonal-anime`
+  let typeClass = `.js-seasonal-anime-list-key-${type2Class[typeString]}`
+
+  // If TVNew or TVCon are selected, filter them out to the specific elements on page
+  if (typeString.substr(0, 2) === 'TV' && typeString !== 'TV') {
+    let tvType = matchSorter(possibleTV, typeString)[0]
+    $(typeClass).children('.anime-header').each(function () {
+      if ($(this).text() === tvType) {
+        classToSearch = $(this).parent().children()
+      }
+    })
+  }
 
   $(classToSearch).each(function () {
-    if (!$(this).hasClass('kids') && !$(this).hasClass('r18')) {
+    if (!$(this).hasClass('kids') && !$(this).hasClass('r18') && $(this)) {
```

Comment:
If you need to check `$(this)`, check it before:
`$(this) && !$(this).hasClass('kids') && !$(this).hasClass('r18')`

But I don't think it's needed?

Reply:
I will check this later, I actually don't think it's needed. Not sure why I added that, think I was testing some stuff and it slipped by. Sorry

Reply:
Should work now, tested and all tests passed without it